### PR TITLE
fix: BasicDealFilter logic

### PR DESCRIPTION
**Pull request by LexLuthr**
Fixes https://github.com/filecoin-project/boost/issues/1208
This PR also extend the deal params exposed to the external deal filters like CIDGravity. Before this an offline deal was sending a false value for IsOffline to external filter.

### CODE REVIEW

File: storagemarket/provider_dealfilter.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func (p *Provider) getDealFilterParams(deal *types.ProviderDealState) (*dealfilt
 		ClientDealProposal: deal.ClientDealProposal,
 		DealDataRoot:       deal.DealDataRoot,
 		Transfer:           deal.Transfer,
+		IsOffline:          deal.IsOffline,
+		SkipIPNIAnnounce:   !deal.AnnounceToIPNI,
+		RemoveUnsealedCopy: !deal.FastRetrieval,
```
Comment (dirkmc) [inline storagemarket/provider_dealfilter.go:51]: Let's do this in a separate PR as it requires a version change on the deal filter JSON object

**Review by dirkmc — APPROVED** [review]

File: storagemarket/provider_dealfilter.go (hunk 1 above)
Comment (LexLuthr) [reply storagemarket/provider_dealfilter.go:51]: Sure. I will remove it with a commit. Running a devnet at the moment to check the original problem. Will push the commit after.
